infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftInfrastructure.java

```diff
@@ -33,28 +32,27 @@
 
   @Inject
   public OpenShiftInfrastructure(
-      EventService eventService,
       OpenShiftRuntimeContextFactory runtimeContextFactory,
       OpenShiftEnvironmentParser envParser,
-      OpenShiftInfrastructureProvisioner infrastructureProvisioner) {
-    super("openshift", ImmutableSet.of("openshift"), eventService);
+      OpenShiftInfrastructureProvisioner infrastructureProvisioner,
+      InstallerRegistry installerRegistry,
+      RecipeRetriever recipeRetriever) {
+    super("openshift", ImmutableSet.of("openshift"), installerRegistry, recipeRetriever);
     this.runtimeContextFactory = runtimeContextFactory;
     this.envParser = envParser;
     this.infrastructureProvisioner = infrastructureProvisioner;
   }
 
   @Override
-  public Environment estimate(Environment environment)
+  public void internalEstimate(InternalEnvironment environment)
       throws ValidationException, InfrastructureException {
     //TODO implement estimation and validation here
```

Comment:
We don't have any issue for that. And I don't think its important enough now. So, I'm removing this todo